Documentation/csiaddons.md

```diff
@@ -0,0 +1,93 @@
+# CSI-Addons for Kubernetes
+
+## Table of Contents
+
+- [CSI-Addons for Kubernetes](#csi-addons-for-kubernetes)
+  - [Table of Contents](#table-of-contents)
+  - [CSI-Addons Controller](#csi-addons-controller)
+    - [Installation by operator](#installation-by-operator)
+  - [Enable the CSI-Addons Sidecar](#enable-the-csi-addons-sidecar)
+  - [Deploying the CSI-Addons Controller](#deploying-the-csi-addons-controller)
+
+## CSI-Addons Controller
+
+The CSI-Addons Controller handles the requests from users to initiate an operation. Users create a CR that the controller inspects, and forwards a request to one or more CSI-Addons side-cars for execution.
+By listing the CSIAddonsNode CRs, the CSI-Addons Controller knows how to connect to the side-cars. By checking the supported capabilities of the side-cars, it can decide where to execute operations that the user requested.
+
+### Installation by operator
+
+A CSI-Addons bundle can be used to install the CSI-Addons Controller with the following steps:
+
+```bash
+$ kubectl create namespace storage-csi-addons
+$ make operator-sdk
+$ ./bin/operator-sdk run bundle -n storage-csi-addons quay.io/csiaddons/k8s-bundle:latest
```

Comment:
There are two methods that upstream rook users are expected to deploy with, either helm or raw manifests. Using operator-sdk is a new approach. What exactly does the bundle install? Or how can we integrate this with the helm chart?

Reply:
Yes, using the operator-sdk is a new approach that is way easier than installing raw crds. The bundle is used used to install the csi-addons controllers. Not sure how do we integrate them with helm. I guess it is fine if we don't. The users can deploy rook using helm and then can use operator-sdk to install the csi-addons controllers. Correct me @nixpanic if I am wrong.

Reply:
If Helm can run a similar command (`operator-sdk` just calls `olm`, I think), the Helm charts should be able to deploy the controller as well. I do not know Helm well enough to help with this.